app/database/operator/server_data_operator/transformers/group.ts

```diff
@@ -42,3 +44,29 @@ export const transformGroupRecord = ({action, database, value}: TransformerArgs)
         fieldsMapper,
     }) as Promise<GroupModel>;
 };
+
+/**
+   * transformGroupMembershipRecord: Prepares a record of the SERVER database 'GroupMembership' table for update or create actions.
+   * @param {TransformerArgs} operator
+   * @param {Database} operator.database
+   * @param {RecordPair} operator.value
+   * @returns {Promise<GroupMembershipModel>}
+   */
+export const transformGroupMembershipRecord = ({action, database, value}: TransformerArgs): Promise<GroupMembershipModel> => {
+    const raw = value.raw as {group_id: string; user_id: string};
+
+    // id of group comes from server response
+    const fieldsMapper = (model: GroupMembershipModel) => {
+        model._raw.id = `${raw.group_id}_${raw.user_id}`;
```

Comment:
other composition ids have a `-` instead of an `_` we should try and be consistent here